src/tools/perf/perftest_rte.h

```diff
@@ -0,0 +1,33 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCX_PERFTEST_RTE_H
+#define UCX_PERFTEST_RTE_H
+
+#include <ucs/type/status.h>
+
+enum perftest_rte_type {
+    PERFTEST_RTE_SOCK = 0,
+    PERFTEST_RTE_MPI,
+    PERFTEST_RTE_MAD,
+    PERFTEST_RTE_MAX
+};
+
+struct perftest_context;
+
+struct perftest_rte {
+    ucs_status_t (*setup)(struct perftest_context *ctx);
+    ucs_status_t (*cleanup)(struct perftest_context *ctx);
+};
+
+extern const struct perftest_rte *rte_list[PERFTEST_RTE_MAX];
```

Comment:
do we really need this global variable?
Maybe it's better to have a function returning perftest_rte?
```
perftest_rte_t * ucp_perftest_find_rte(const ucp_perftest_context_t *ctx)
{
    if (ctx->mad_port != NULL) {
        return &plugin_mad_rte;
    } else if (ctx->mpi) {
        return &plugin_mpi_rte;
    } else {
        return &plugin_sock_rte;
    }
}

Reply:
variable needs to be exposed to mad .so plugin